src/DocumentEditor.ts

```diff
@@ -20,19 +20,35 @@ export class DocumentEditor implements vscode.Disposable {
 
     private readonly dontShowKey: string = 'cosmosDB.dontShow.SaveEqualsUpdateToAzure';
 
-    public async showDocument(docNode: IEditableNode): Promise<void> {
-        const localDocPath = path.join(os.tmpdir(), randomUtils.getRandomHexString(12), 'cosmos-editor.json');
+    public async showDocument(docNode: IEditableNode, extensionPath: string, fileName: string): Promise<void> {
+        const localDocPath = path.join(extensionPath, fileName);
         await fse.ensureFile(localDocPath);
 
         const document = await vscode.workspace.openTextDocument(localDocPath);
-        this.fileMap[localDocPath] = [document, docNode];
+        if (!(localDocPath in this.fileMap)) {
+            this.fileMap[localDocPath] = [document, docNode];
```

Comment:
Don't you need to do this if the user selects 'yes' in the else case below? I would just move this line down to 41